jormungandr/src/fragment/pool.rs

```diff
@@ -67,40 +71,52 @@ impl Pool {
             .zip(fragments_exist_in_logs)
             .filter(|(_, exists_in_logs)| !exists_in_logs)
             .map(|(fragment, _)| fragment);
-        let new_fragments = self.pool.insert_all(new_fragments);
-        let count = new_fragments.len();
-        debug!(
-            self.logger,
-            "{} of the received fragments were added to the pool", count
-        );
-        let fragment_logs = new_fragments
-            .iter()
-            .map(move |fragment| FragmentLog::new(fragment.id(), origin))
-            .collect::<Vec<_>>();
-        for fragment in new_fragments.into_iter() {
-            let fragment_msg = NetworkMsg::Propagate(PropagateMsg::Fragment(fragment));
-            network_msg_box
-                .send(fragment_msg)
-                .await
-                .map_err(Error::CannotPropagate)?;
+
+        let mut max_added = 0;
+
+        for (i, pool) in self.pools.iter_mut().enumerate() {
+            let new_fragments = pool.insert_all(new_fragments.clone());
+            let count = new_fragments.len();
+            debug!(
+                self.logger,
+                "{} of the received fragments were added to the pool number {}", count, i
+            );
+            let fragment_logs = new_fragments
+                .iter()
+                .map(move |fragment| FragmentLog::new(fragment.id(), origin))
+                .collect::<Vec<_>>();
+            for fragment in new_fragments.into_iter() {
+                let fragment_msg = NetworkMsg::Propagate(PropagateMsg::Fragment(fragment));
+                network_msg_box
+                    .send(fragment_msg)
+                    .await
+                    .map_err(Error::CannotPropagate)?;
```

Comment:
Let's give some thought to not propagating fragments redundantly. This should be outside of the loop over the pools.